promql/engine_test.go

```diff
@@ -3230,12 +3270,45 @@ func TestSparseHistogram_HistogramCountAndSum(t *testing.T) {
 	require.Len(t, vector, 1)
 	require.Nil(t, vector[0].H)
 	require.Equal(t, h.Sum, vector[0].V)
+
+	// float histogram
+	ts = int64(11 * time.Minute / time.Millisecond)
```

Comment:
This test feels a little fragile. Because both integer and float histogram samples are in the same series and close by. How about we use the same loop like other tests and if-else where required (i.e. AppendHistogram and when checking the result)